ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Inspect.hs

```diff
@@ -8,166 +8,52 @@
-    invertMap = Map.fromListWith (<>) . fmap swizzle . Map.toList
-      where
-        swizzle (a, b) = (b, [a])
-
-    -- Updated proposed within the proposal window
-    proposals :: Map (SL.KeyHash 'SL.Genesis (EraCrypto era)) (Core.PParamsUpdate era)
-    SL.ProposedPPUpdates proposals =
-          fromMaybe SL.emptyPPPUpdates
-        . Core.getProposedPPUpdates
-        . view SL.newEpochStateGovStateL
-        . shelleyLedgerState
-        $ st
-
-    -- A proposal is accepted if the number of votes is equal to or greater
-    -- than the quorum. The quorum itself must be strictly greater than half
-    -- the number of genesis keys, but we do not rely on that property here.
-    quorum :: Word64
-    quorum = SL.sgUpdateQuorum genesis
-
-    -- The proposals in 'SL.proposals' are for the upcoming epoch
-    -- (we ignore 'futureProposals')
-    currentEpoch :: EpochNo
-    currentEpoch = SL.nesEL . shelleyLedgerState $ st
-
-{-------------------------------------------------------------------------------
-  Inspection
--------------------------------------------------------------------------------}
-
 data ShelleyLedgerUpdate era =
-    ShelleyUpdatedProtocolUpdates [ProtocolUpdate era]
+  ShelleyUpdatedPParams (Maybe (Core.PParams era)) EpochNo
 
-deriving instance Eq (Core.PParamsUpdate era) => Eq (ShelleyLedgerUpdate era)
-deriving instance Show (Core.PParamsUpdate era) => Show (ShelleyLedgerUpdate era)
+deriving instance Eq (Core.PParams era) => Eq (ShelleyLedgerUpdate era)
+deriving instance Show (Core.PParams era) => Show (ShelleyLedgerUpdate era)
 
-instance Show (Core.PParamsUpdate era) => Condense (ShelleyLedgerUpdate era) where
+instance Show (Core.PParams era) => Condense (ShelleyLedgerUpdate era) where
   condense = show
 
 instance ShelleyBasedEra era => InspectLedger (ShelleyBlock proto era) where
   type LedgerWarning (ShelleyBlock proto era) = Void
   type LedgerUpdate  (ShelleyBlock proto era) = ShelleyLedgerUpdate era
 
-  inspectLedger tlc before after = do
+  inspectLedger _tlc before after = do
       guard $ updatesBefore /= updatesAfter
-      return $ LedgerUpdate $ ShelleyUpdatedProtocolUpdates updatesAfter
+      return $ LedgerUpdate updatesAfter
     where
-      genesis :: SL.ShelleyGenesis (EraCrypto era)
-      genesis = shelleyLedgerGenesis (configLedger tlc)
 
-      updatesBefore, updatesAfter :: [ProtocolUpdate era]
-      updatesBefore = protocolUpdates genesis before
-      updatesAfter  = protocolUpdates genesis after
+      updatesBefore, updatesAfter :: ShelleyLedgerUpdate era
+      updatesBefore = pparamsUpdate before
+      updatesAfter  = pparamsUpdate after
+
+pparamsUpdate ::
+       forall era proto. ShelleyBasedEra era
+    => LedgerState (ShelleyBlock proto era)
+    -> ShelleyLedgerUpdate era
+pparamsUpdate st =
+    let nes = shelleyLedgerState st
+     in ShelleyUpdatedPParams
+          (nes ^. SL.newEpochStateGovStateL . SL.futurePParamsGovStateG)
```

Comment:
If want to further minimize the API surface: It would suffice to just store the protocol version here.

Reply:
EpochNo is used in `shelleyTransition`, but my worry was that `guard $ updatesBefore /= updatesAfter` is affected by removal of this epochNo, so if they are guaranteed to be the same, then we can remove it from `ShelleyLedgerUpdate`

Reply:
Right, I was imprecise; I think it makes sense to keep `EpochNo` here; but the `Maybe (Core.PParams era)` could be changed to `Maybe ProtVer`.

Reply:
So this suggestion would be fine if `inspectLedger` was only used for predicting protocol version. I am just really not sure that this is indeed the case. I would suspect that updates to other protocol parameters could be useful for inspection as well?

In other words if I thin this to only `ProtVer` then `inspecLedger` will return `Nothing` on any other parameter update, while previously it would return a `Just` value, due to some parameter change and this `guard $ updatesBefore /= updatesAfter`